fixed: all add-ons got duplicated in the addon table

opdenkamp opened the pull request:
PR https://github.com/xbmc/xbmc/pull/6568 deleted all repo entries from the db, but didn't delete the add-ons. This leads to a dupe of each add-on when upgrading from an older version, as the repo gets inserted another time, including all it's add-ons.

Fixed by defaulting to version 0.0.0, which just triggers an update.

ping @tamland @mkortstiege @MartijnKaijser

opdenkamp committed:
fixed: all add-ons got duplicated in the addon table after b72f00a75c63ebe5b846d28824bcbe3ba1daebc6

opdenkamp commented:
this won't fix the issue for anyone who upgraded in the meantime by running a nightly or compiling from source. those will have to delete `~/.kodi/userdata/Database/Addons17.db`

tamland commented:
I don't think it *certainly* create duplicates though. I haven't been able to reproduce it so far, I'm not sure what the exact conditions are, but I guess that line is still wrong. Can't you rather bump version and add a special case for v17 that clears the `addons` table? So people don't have to delete it manually

opdenkamp commented:
sure, will add the bump.
The conditions are starting Helix, do a refresh of the add-ons if it doesn't happen automatically, then close it. Then start current master and do the same thing, and go to "Get add-ons" and pick a category. This resulted in a dupe of each add-on here, and afaict this will happen always and for everyone.

opdenkamp committed:
fixed: don't return local add-ons from CAddonDatabase::GetAddons()

opdenkamp commented:
updated the PR to remove the dupes, but only if the user was running db version 17.
fixed the local add-ons showing up in the get add-ons one.

jenkins build this please

xhaggi commented on xbmc/addons/AddonDatabase.cpp:
```diff
@@ -324,7 +328,7 @@ bool CAddonDatabase::GetAddons(VECADDONS& addons, const ADDON::TYPE &type /* = A
     if (NULL == m_pDS2.get()) return false;
 
-    std::string sql = PrepareSQL("select distinct addonID from addon");
+    std::string sql = PrepareSQL("SELECT DISTINCT a.addonID FROM addon a, addonlinkrepo b WHERE b.idRepo > 0 AND a.id = b.idAddon");
```
should we use implicit inner join here which makes it cleaner to read?

opdenkamp replied:
personally I find this easier to read that a join, but I didn't want to be changing anything in here really, just fix the bug that I saw, so feel free to change it ;)

opdenkamp commented:
build failure unrelated
